gobblin-yarn/src/main/java/gobblin/yarn/YarnService.java

```diff
@@ -340,6 +369,60 @@ private String buildContainerCommand(Container container) {
   }
 
   /**
+   * Check the exit status of a completed container and see if the replacement container
+   * should try to be started on the same node. Some exit status indicates a disk or
+   * node failure and in such cases the replacement container should try to be started on
+   * a different node.
+   */
+  private boolean shouldStickToTheSameNode(int containerExitStatus) {
+    switch (containerExitStatus) {
+      case ContainerExitStatus.DISKS_FAILED:
```

Comment:
Yes, it falls through to `ContainerExitStatus.ABORTED` in which case it returns `false`.